src/server/frontend_wayland/generated/wlr-layer-shell-unstable-v1_wrapper.cpp

```diff
@@ -14,6 +14,27 @@
 
 #include "mir/log.h"
 
+namespace
+{
+void internal_error_processing_request(struct wl_client* client, std::string const& method_name)
+{
+#if (WAYLAND_VERSION_MAJOR > 1 || (WAYLAND_VERSION_MAJOR == 1 && WAYLAND_VERSION_MINOR > 16))
+    wl_client_post_implementation_error(
+        client,
+        "Mir internal error processing %s request",
+        method_name.c_str(),
+        "Exception processing " + method_name + " request");
+#else
+    wl_client_post_no_memory(client);
+#endif
+    ::mir::log(
+        ::mir::logging::Severity::critical,
```

Comment:
"critical" should mean something so serious the server internal state is messed up and the only safe action is to shut down.

Failing to process an invalid client request is probably "informational", certainly no more than "warning".

(I recognise this is pre-existing in this code, but this seems a good time to address it.)

Reply:
This codepath should only be hit by failing to process a *valid* client request due to a Mir bug. “Critical” might be too high, but “warning” is too low.

Reply:
OK. The description could make this clear.

Assuming that this is indeed reporting a postcondition failure (and not a precondition failure nor an "expected" failure mode) then I agree that "Severity::error" is the correct thing to log if we're otherwise "eating" the exception.

Reply:
Changed to `error`